dataset.d.ts

```diff
@@ -192,13 +185,13 @@ export interface Dataset<OutQuad extends BaseQuad = Quad, InQuad extends BaseQua
      */
     union(quads: Dataset<InQuad>): Dataset<OutQuad, InQuad>;
 
-    match(subject?: Term | null, predicate?: Term | null, object?: Term | null, graph?: Term | null): Dataset<OutQuad, InQuad>;
+    match(subject?: InQuad['subject'] | TermPattern, predicate?: InQuad['predicate'] | TermPattern, object?: InQuad['object'] | TermPattern, graph?: InQuad['graph'] | TermPattern): Dataset<OutQuad, InQuad>;
```

Comment:
This change allows the implementor to narrow the subtype of Quad they will accept, for example in a dataset that does not support RDF-Star. I don't see what makes this incompatible with the spec. Doesn't this conflict with what you said above?
> Implementations may or may not require their specific types being used.

Reply:
Right, I see how it can appear contradictory.

While the vectors of the changes are opposite, in both cases I think they risk breaking the current behaviour. As mentioned above, a `Dataset` accepts any term in the `match` and other methods. Making it more strict will inevitable break the type checks in many projects.

I thought that maybe your changes could work if instead the defaults for type arguments were

```diff
-OutQuad extends BaseQuad = Quad
+OutQuad extends BaseQuad = BaseQuad
```

Thus, defaulting to `Term` for all parameters. However, consider `type DatasetExt = DatasetCore<QuadExt>`. Same thing happens, that TypeScript will refuse to transpile when `Term` is used as argument but `rdf-ext` will happily ignore "wrong" terms...